lib/import/colorectal/providers/leeds/leeds_handler_colorectal.rb

```diff
@@ -1,696 +1,441 @@
-
-          ABSENT_REGEX = /is absent in this patient/i .freeze
-          NO_DEL_REGEX = /this patient does not have the deletion/i .freeze
-          NOMUT_REGEX = /(?<notmut>No mutations were identified in|this patient does not have|has not identified any mutations in this patient|no mutation has been identified|no evidence of a deletion or duplication)|not detected in this patient|did not identify/i .freeze
-          EXON_ALTERNATIVE_REGEX = /(heterozygous)?[\s\w]{2,} (?<pathogenic>pathogenic)? (?<genes>APC|BMPR1A|EPCAM|MLH1|MSH2|MSH6|MUTYH|PMS2|POLD1|
-                                    POLE|PTEN|SMAD4|STK11|GREM1|NTHL1) (?<delmut>deletion|mutation|inversion|duplication)[\s\w]{2,}(?<exons>exon(s).[0-9]+(.?)[0-9]+?)|
-                                    (pathogenic)? (?<genes>APC|BMPR1A|EPCAM|MLH1|MSH2|MSH6|MUTYH|PMS2|POLD1|
-                                    POLE|PTEN|SMAD4|STK11|GREM1|NTHL1)[\w\s]{0,2}(?<invdupdel>inversion|duplication|deletion)\s[\w\s]+{0,2}(?<exons>exon(s).[0-9]+(.?)[0-9]+?)|
-                                    (pathogenic)? (?<invdupdel>inversion|duplication|deletion)\s[\w\s]+{0,2}(?<exons>exon(s).[0-9]+(.?)[0-9]+?) ([\w\s]+)?
-                                    (?<genes>APC|BMPR1A|EPCAM|MLH1|MSH2|MSH6|MUTYH|PMS2|POLD1|
-                                    POLE|PTEN|SMAD4|STK11|GREM1|NTHL1)[\w\s]{0,2}?|(familial)? pathogenic deletion of 
-                                    (?<genes>APC|BMPR1A|EPCAM|MLH1|MSH2|MSH6|MUTYH|PMS2|POLD1|
-                                    POLE|PTEN|SMAD4|STK11|GREM1|NTHL1) (?<delmut>deletion|mutation|inversion|duplication)[\s\w]{2,}
-                                    (?<exons>exon(s).[0-9]+(.?)[0-9]+?)|(familial)? pathogenic (?<genes>APC|BMPR1A|EPCAM|MLH1|MSH2|MSH6|MUTYH|PMS2|POLD1|
-                                    POLE|PTEN|SMAD4|STK11|GREM1|NTHL1) (?<delmut>deletion|mutation|inversion|duplication)/ix .freeze
+          include Import::Helpers::Colorectal::Providers::Rr8::Constants
 
           def initialize(batch)
-            @extractor = ReportExtractor::GenotypeAndReportExtractor.new
-            @negative_test = 0 # Added by Francesco
-            @positive_test = 0 # Added by Francesco
+            @genes_hash = YAML.safe_load(File.open(Rails.root.join(GENES_FILEPATH)))
+            @status_hash = YAML.safe_load(File.open(Rails.root.join(STATUS_FILEPATH)))
             super
           end
 
-          # rubocop:disable Metrics/AbcSize
           def process_fields(record)
+            populate_variables(record)
+            return unless should_process
+
+            populate_and_persist_genotype(record)
+          end
+
+          # populate class variables that can be used over different methods
```

Comment:
I think these are instance variables, may be worth changing the comment

Reply:
Thanks @ollietulloch , done in latest commit .